src/task.rs

```diff
@@ -53,14 +57,17 @@ impl Task {
             slots: Vec::new(),
             confirmed_start: None,
             confirmed_deadline: None,
-            internal_index: 0,
+            internal_marker,
         }
     }
 
     //TODO: The current way this is done may not be entirely accurate for tasks that can be done on
     //multiple days within certain time bounds.
```

Comment:
Shouldn't all constraints be applied to the Slot vec at Task initialization - leaving only the actually valid Slots?

Reply:
I mean that when a Task gets constructed it sgould not be allowed to be in an invalid state. Initializing it with Task::new before knowing which vec<Slot> correspond to its time constraints will make this Task's state invalid.

Reply:
Just to help me understand which part of the process you're referring to, let me recap how it's currently working:

1) The task generator generates tasks from goals with correct constraints based on the goal. i.e. the tasks are given correct start,deadline, and aftertime,beforetime values. These tasks initially have no slots assigned though.
2) Each task is assigned slots based on it's constraints.
3) Each task is scheduled.

Are you referring to what happens in step 1?

Reply:
Thanks for the breakdown.
That makes communucation easy.

What I'm referring to is making 2 part of 1. 

Allowing the Tasks to be created in 1 and then 'assigning' (the right word for this case) slots would be OK if you create an extra Task status 'NOSLOTASSIGNEDYET' - but it's easier to not allow Tasks with an inconsistent internal state.

Reply:
Okay I see now. Will consider this after resolving next couple of issues. After completing #159